ci/ray_ci/doc/autodoc.py

```diff
@@ -55,11 +56,28 @@ def _get_autodoc_rsts(self) -> List[str]:
         if self._autodoc_rsts is not None:
             return self._autodoc_rsts
 
-        dir = os.path.dirname(self._head_rst_file)
-        self._autodoc_rsts = [self._head_rst_file]
-        with open(self._head_rst_file, "r") as f:
+        self._autodoc_rsts = {self._head_rst_file}
+        visited = set()
+        while len(self._autodoc_rsts) > len(visited):
+            not_visited = self._autodoc_rsts - visited
+            for rst in not_visited:
+                visited.add(rst)
+                self._autodoc_rsts.update(self._get_autodoc_rsts_from_single_file(rst))
+
+        return self._autodoc_rsts
+
+    def _get_autodoc_rsts_from_single_file(self, rst_file: str) -> Set[str]:
```

Comment:
maybe add a unit test for this new file parsing function?

and maybe just call it `_get_autodoc_rsts_in_file` ? feels that it does not have to emphasize "single"